scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -441,150 +428,106 @@ public long getRunningJobsCount() {
         });
     }
 
     public long getTotalTasksCount() {
-        return executeReadOnlyTransaction(new SessionWork<Long>() {
-
-            @Override
-            public Long doInTransaction(Session session) {
-                Query query = session.getNamedQuery("getTotalTasksCount");
-                Long count = (Long) query.uniqueResult();
-                return count;
-            }
-
+        return executeReadOnlyTransaction(session -> {
+            Query query = session.getNamedQuery("getTotalTasksCount");
+            Long count = (Long) query.uniqueResult();
+            return count;
         });
     }
 
     public double getMeanJobPendingTime() {
-        return executeReadOnlyTransaction(new SessionWork<Double>() {
-            @Override
-            public Double doInTransaction(Session session) {
-                Query query = session.getNamedQuery("getMeanJobPendingTime");
-                Double result = (Double) query.uniqueResult();
-                return result == null ? 0 : result;
-            }
-
+        return executeReadOnlyTransaction(session -> {
+            Query query = session.getNamedQuery("getMeanJobPendingTime");
+            Double result = (Double) query.uniqueResult();
+            return result == null ? 0 : result;
         });
     }
 
     public double getMeanJobExecutionTime() {
-        return executeReadOnlyTransaction(new SessionWork<Double>() {
-            @Override
-            public Double doInTransaction(Session session) {
-                Query query = session.getNamedQuery("getMeanJobExecutionTime");
-                Double result = (Double) query.uniqueResult();
-                return result == null ? 0 : result;
-            }
+        return executeReadOnlyTransaction(session -> {
+            Query query = session.getNamedQuery("getMeanJobExecutionTime");
+            Double result = (Double) query.uniqueResult();
+            return result == null ? 0 : result;
         });
     }
 
     public double getMeanJobSubmittingPeriod() {
-        return executeReadOnlyTransaction(new SessionWork<Double>() {
-            @Override
-            public Double doInTransaction(Session session) {
-                Query query = session.getNamedQuery("getMeanJobSubmittingPeriod");
-                Object[] result = (Object[]) query.uniqueResult();
-                Long count = (Long) result[0];
-                Long minSubmittedTime = (Long) result[1];
-                Long maxSubmittedTime = (Long) result[2];
-                if (count < 2) {
-                    return 0d;
-                } else {
-                    return (maxSubmittedTime - minSubmittedTime) / (double) (count - 1);
-                }
+        return executeReadOnlyTransaction(session -> {
+            Query query = session.getNamedQuery("getMeanJobSubmittingPeriod");
+            Object[] result = (Object[]) query.uniqueResult();
+            Long count = (Long) result[0];
```

Comment:
Is it supposed to be able to throw an IndexOutOfBoundsException here?

Reply:
I don't know.